### DISCUSSION TITLE
Why 'ccache -V' shows 'ccache version unknown' message?

### DISCUSSION THREAD

**Discussion by sameermahant** [ccache/ccache#418] [Support]
### Why 'ccache -V' shows 'ccache version unknown' message? ###
Downloaded the **ccache v3.7.1** from https://github.com/ccache/ccache/releases/download/v3.7.1/ccache-3.7.1.tar.xz

Installed it successfully, but `ccache -V` command report wrong version:
``` bash
[root@localhost ~]# /opt/toolchain/ccache-3.7.1/bin/ccache -V
ccache version unknown
```

Also looked at related issue [ccache --version reports "version unkown"](https://github.com/ccache/ccache/issues/225), but don't think this issue is similar to it.

I observed that running `/configure --prefix=/opt/toolchain/ccache-3.7.1` modified the `src/version.c` file contents.
##### Before: 
```c
extern const char CCACHE_VERSION[]; const char CCACHE_VERSION[] = "3.7.1";
```

##### After:
```c
extern const char CCACHE_VERSION[]; const char CCACHE_VERSION[] = "unknown";
```
---

**Is it a bug or environment issue specific to my setup?**

---

##### GCC Compiler version: 4.8.5-36
```bash
[root@localhost ~]# gcc --version
gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
##### G++ Compiler version: 4.8.5-36
```bash
[root@localhost ~]# g++ --version
g++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

##### Operating System: Cent OS 7.6 (minimal install)
```bash
[root@localhost ~]# uname -a
Linux localhost.localdomain 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

##### Actual behavior:
```bash
[root@localhost ~]# /opt/toolchain/ccache-3.7.1/bin/ccache -V
ccache version unknown

Copyright (C) 2002-2007 Andrew Tridgell
Copyright (C) 2009-2019 Joel Rosdahl

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation; either version 3 of the License, or (at your option) any later
version.
```

##### Expected behavior:
```bash
[root@localhost ~]# /opt/toolchain/ccache-3.7.1/bin/ccache -V
ccache version 3.7.1

Copyright (C) 2002-2007 Andrew Tridgell
Copyright (C) 2009-2019 Joel Rosdahl

This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation; either version 3 of the License, or (at your option) any later
version.
```

##### Steps to reproduce:
```bash
wget -c https://github.com/ccache/ccache/releases/download/v3.7.1/ccache-3.7.1.tar.xz
tar -xvf ccache-3.7.1.tar.xz
cd ccache-3.7.1
./autogen.sh
./configure --prefix=/opt/toolchain/ccache-3.7.1
make
make install

/opt/toolchain/ccache-3.7.1/bin/ccache -V
```

**Comment by afbjorklund** [comment]
When you run `./autogen.sh`, it will regenerate files from the git information. You don't need this step.

> I observed that running /configure --prefix=/opt/toolchain/ccache-3.7.1 modified the src/version.c file contents.

Seems that it was "autogen.sh" that modified the file... ?

``` sh
git describe --dirty 2>/dev/null || echo vunknown
```

**Comment by sameermahant** [comment]
As I was creating a minimal development tool-chain / environment, I did not had **git** installed.
I already seen the code that generate version using some git command, so installed git and retried the installation. The result was same i.e. **unknown version**.

I already checked at which step `src/version.c` file was modified. It was only after I executed the `./configure` command.

I will try the above mentioned git command tomorrow and will share the result.
If the result repeats, please suggest some files which I can look into. If possible I will try to fix it.

Thanks for the reply.

**Comment by sameermahant** [comment]
As I am downloading the `ccache-3.7.1.tar.xz` and not cloning the git branch, don't think the git meta info will be there.

![CCache_version_unknown](https://user-images.githubusercontent.com/3286086/57477616-b0f56f80-72b6-11e9-8691-c6ab18d07073.png)

**Comment by afbjorklund** [comment]
When you download the tarball, just **skip** the `./autogen.sh` step (intended for developers)

See the [INSTALL.md](https://github.com/ccache/ccache/blob/master/doc/INSTALL-from-release-archive.md) file, in the release tarball

**Comment by sameermahant** [comment]
It worked.
Thanks.

### Steps to install ccache from release version
```bash
> wget -c https://github.com/ccache/ccache/releases/download/v3.7.1/ccache-3.7.1.tar.xz
> tar -xvf ccache-3.7.1.tar.xz
> cd ccache-3.7.1

> ./configure
> make
> make install
```
